packages/dashboard/src/components/alert-store.tsx

```diff
@@ -17,90 +19,49 @@ export interface AlertToDisplay extends RobotWithTask {
   show: boolean;
 }
 
-const returnOnlyRobotsWithTask = (value: Record<string, RobotWithTask[]>) => {
-  return Object.values(value)
-    .flatMap((r) => r)
-    .filter((element) => {
-      if (element.task) {
-        return true;
-      }
-      return false;
-    });
-};
-
 export const AlertStore = React.memo(() => {
   const rmf = React.useContext(RmfAppContext);
-
-  const [fleets, setFleets] = React.useState<string[]>([]);
-  const [robotsWithTasks, setRobotsWithTasks] = React.useState<Record<string, RobotWithTask[]>>({});
+  const [taskAlerts, setTaskAlerts] = React.useState<Record<string, Alert>>({});
+  const [robotAlerts, setRobotAlerts] = React.useState<Record<string, Alert>>({});
 
   React.useEffect(() => {
     if (!rmf) {
       return;
     }
-
-    const sub = rmf.fleetsObs.subscribe((fleets) => {
-      setFleets(
-        fleets.reduce<string[]>((acc, f) => {
-          if (f.name) {
-            acc.push(f.name);
-          }
-          return acc;
-        }, []),
-      );
+    const sub = rmf.alertObsStore.subscribe(async (alert) => {
+      if (alert.category === 'task') {
```

Comment:
Does it make sense for us to create a [Typescript enum](https://www.typescriptlang.org/docs/handbook/enums.html#string-enums) if the api client didn't provide one for [categories](https://github.com/open-rmf/rmf-web/pull/694/files#diff-ba1b98e67ee0daeb8df03b15ead4fd9cb68dc27678160ac40cec2aa24c3ef004R13-R17)?

Reply:
Yup, good idea. Ideally it would have come from the API, but alas since we are using the pydantic creator from tortoise, we don't get the luxury of handling enums from API like `TaskState.Status`. I've made the changes https://github.com/open-rmf/rmf-web/pull/694/commits/207bd8d94f7d6845ef7f2c28ab3843eb6824f1e1, and commented that it needs to adhere to the tortoise model